include/aspect/fe_variable_system.h

```diff
@@ -0,0 +1,423 @@
+/*
+  Copyright (C) 2016 by the authors of the ASPECT code.
+
+  This file is part of ASPECT.
+
+  ASPECT is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2, or (at your option)
+  any later version.
+
+  ASPECT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file doc/COPYING.  If not see
+  <http://www.gnu.org/licenses/>.
+*/
+
+#ifndef __aspect__fe_variable_system_h
+#define __aspect__fe_variable_system_h
+
+#include <aspect/global.h>
+
+namespace dealii
+{
+  /**
+   * Structure to describe everything necessary to define a single variable of
+   * the class FEVariableSystem. It groups the FiniteElement<dim> with other
```

Comment:
I think what would be useful to say here at the top is that this class defines *a variable in isolation*, whereas FEVariableSystem defines a variable *as part of a collection of variables in an FESystem*. Can you add some such sentence to the documentation?

Realizing this, it also becomes clear to me why I am so confused about the n_blocks variable, since that is in reality something that is not specific to a variable in isolation, but in connection to other variables. Would a better strategy be to not let this member variable be part of the current class, but pass it as a second, independent, argument when combining VariableDefinition objects into a system?